code/game/objects/items.dm

```diff
@@ -165,7 +166,11 @@
 	else
 		if(isliving(src.loc))
 			return
-	user.put_in_active_hand(src)
+	if(user.put_in_active_hand(src))
+		if (isturf(old_loc))
+			var/obj/effect/temp_visual/item_pickup_ghost/ghost = new(old_loc, src)
```

Comment:
In combination with the above, change this line to var/obj/effect/temp_visual/item_pickup_ghost/ghost = new(get_turf(old_loc), src)